yt_dlp/extractor/twitch.py

```diff
@@ -179,6 +179,14 @@ def _download_access_token(self, video_id, token_kind, param_name):
             video_id, ops,
             'Downloading %s access token GraphQL' % token_kind)['data'][method]
 
+    def _get_thumbnails(self, thumbnail):
+        return [{
+            'url': re.sub(r'\d+x\d+\.jpg', '0x0.jpg', thumbnail),
```

Comment:
This causes problems with custom VOD thumbnails, since they do not have to be JPEGs, as such a more flexible RegEx should be used.

**Example:**
```py
'url': re.sub(r'\d+x\d+\.(jpg|jpeg|png|gif|webp)', '0x0.\g<1>', thumbnail),
```

> This would also be more future-proof in-case Twitch changes their thumbnail extension.

Reply:
Might as well do `\.\w+` then

Reply:
> Might as well do `\.(\w+)` then

That seems a bit overly broad and could lead to edge cases matching strings that shouldn't. At the very least it should then be limited to a maximum amount of characters that would be reasonable for an image extension string. Overall a set of predetermined valid extensions would be a safer bet.

As such if you really would want to use `\w`, which I can't fully recommend, I'd suggest something closer to `\w{3,5}`, since this would at least limit it to a reasonable image extension size. However even using `\w{3,5}` could lead to edge cases that should mostly be avoided using a predetermined set of valid extensions.

While my suggested RegEx change isn't perfect, the amount of potential edge cases it can raise is fairly low and it remains simple and readable. To achieve the safest option a more complex RegEx string which would check that the size argument and image extension are at the end of the string using `$` and accounts for the edge case where a query string `?` or a URI fragment `#` might be included instead of the image extension being at the end.

**Example of more complex RegEx:**
```py
'url': re.sub(r'\d+x\d+\.([a-zA-Z]{3,5})([?#]|$)', '0x0.\g<1>\g<2>', thumbnail),
```
This RegEx makes sure that the size argument and image extension are at the end of the URL and accounts for edge cases where URL queries or URI fragments are included before the end of the string. This RegEx also omits the use of `\w` for `[a-zA-Z]`, since `\w` would match invalid image extension characters like `_`. This becomes quite a bit more complex than I feel is needed, as such I again reiterate that a set of predetermined valid extensions would be simpler and achieve a fairly comparable level of safety against edge cases.

Reply:
The `thumbnail` var is already expected to be a valid thumbnail URL. The only objective of the regex is to replace the resolution, not to validate the URL. I do agree with checking the end though
```suggestion
            'url': re.sub(r'\d+x\d+(\.\w+)($|(?=[?#]))', '0x0\g<1>', thumbnail),
```

Reply:
Alright, I'm happy with the suggested change, thank you